src/components/notifications-table/NotificationsTable.jsx

```diff
@@ -41,35 +41,36 @@ class NotificationsTable extends React.Component {
         this.props.history.push("/");
     };
 
-    deleteNotification = notification => {
-        this.setState({ toDelete: notification });
+    deleteNotification = notifications => {
+        this.setState({ toDelete: notifications });
     };
 
     cancelDelete = () => {
         this.setState({ toDelete: null });
     };
 
     confirmDelete = async () => {
+        const { loading, d2 } = this.props;
         const { toDelete } = this.state;
 
-        const syncReport = new SyncReport(toDelete);
-        this.setState({ toDelete: null });
-        this.props.loading.show(true, i18n.t("Deleting notification"));
-
-        await syncReport.remove(this.props.d2).then(response => {
-            this.props.loading.reset();
-
-            if (response.status) {
-                this.props.snackbar.success(i18n.t("Deleted {{name}}", { name: toDelete.name }));
-
-                // TODO: Workaround, add a way to force render of ObjectsTable on-demand
-                this.setState({ tableKey: Math.random() });
-            } else {
-                this.props.snackbar.error(
-                    i18n.t("Failed to delete {{name}}", { name: toDelete.name })
-                );
-            }
-        });
+        loading.show(true, i18n.t("Deleting Notifications"));
+        const notifications = toDelete.map(data => new SyncReport(data));
+
+        const results = [];
+        for (const notification of notifications) {
+            results.push(await notification.remove(d2));
+        }
```

Comment:
Whenever we have a pmap-like abstraction at hand: `const results = await promiseMap(notifications, notification => notification.remove(d2);)`, I think there is none now in the projcet.

Reply:
Yeah looks good when we implement race protection on data store level.